tests/integration/Owncloud/OcisSdkPhp/OcisTest.php

```diff
@@ -135,4 +138,99 @@ public function testCreateDriveInvalidQuota(int $quota): void
         $this->assertCount($countDrivesAtStart, $ocis->listMyDrives());
     }
+
+    public function testListGroups(): void
+    {
+        $token = $this->getAccessToken('admin', 'admin');
+        $ocis = new Ocis($this->ocisUrl, $token, ['verify' => false]);
+        $groups = $ocis->listGroups();
+        $this->assertCount(self::GROUP_COUNT, $groups);
+        $this->assertContainsOnlyInstancesOf(Group::class, $groups);
+        foreach ($groups as $group) {
+            $this->assertIsString($group->getId());
+            $this->assertIsString($group->getDisplayName());
+            $this->assertIsArray($group->getGroupTypes());
+            $this->assertIsArray($group->getMembers());
+        }
+    }
+    /**
+     * @return array<int,array<int,string>>
+     */
+    public function expandData(): array
+    {
+        return [
+            ["users"],
+        ];
+    }
+    /**
+     * @dataProvider expandData
+     *
+     * @return void
+     */
+    public function testListGroupsExpanded()
+    {
+        $token = $this->getAccessToken('admin', 'admin');
+        $ocis = new Ocis($this->ocisUrl, $token, ['verify' => false]);
+        $groups = $ocis->listGroups(expandMembers: true);
+        $this->assertCount(self::GROUP_COUNT, $groups);
+        foreach ($groups as $group) {
+            $this->assertGreaterThan(0, count($group->getMembers()));
+        }
+    }
+    /**
+     * @return array<int,array<int,string|int>>
+     */
+    public function searchText(): array
+    {
+        return [
+            ["asdas", 0, ""],
+        ];
+    }
+    /**
+     * @dataProvider searchText
+     *
+     * @param string $searchText
+     * @param integer $count
+     * @param string $groupDisplatName
+     * @return void
+     */
+    public function testListGroupSerch(string $searchText, int $count, string $groupDisplatName)
+    {
+        $token = $this->getAccessToken('admin', 'admin');
+        $ocis = new Ocis($this->ocisUrl, $token, ['verify' => false]);
+        $groups = $ocis->listGroups(search: $searchText);
+        $this->assertCount($count, $groups);
+        foreach ($groups as $group) {
+            $this->assertSame($groupDisplatName, $group->getDisplayName());
+        }
+    }
+    /**
+     * @return array<int, array<int, array<int,  string>|int|OrderDirection|string>>
+     */
+    public function orderDirection(): array
+    {
+        return [
+            [OrderDirection::ASC, 0, "ph", ["philosophy-haters", "physics-lovers"]],
+            [OrderDirection::DESC, 0, "ph", ["physics-lovers", "philosophy-haters"]],
+        ];
+    }
+    /**
+     * @dataProvider orderDirection
+     *
+     * @param OrderDirection $orderDirection
+     * @param int $count
+     * @param string $searchText
+     * @param array<int,string> $resultGroups
+     * @return void
+     */
+    public function testListGroupSort(OrderDirection $orderDirection, int $count, string $searchText, array $resultGroups)
```

Comment:
`$count` should not be needed, the count is the amount of items in `$resultGroups`